### ISSUE TITLE
`instanceof Uint8Array` check flaky when using `jsdom` environment

### ISSUE THREAD

**Issue by juliusmarminge** [vitest-dev/vitest#5183]
### Describe the bug

It seems like when the jsdom environment is used, some `Uint8Array` checks are flaky and some returns true while other returns false. This leads to tests failing when they shouldn't.

---

> [!NOTE] 
> This issue came to be from: https://github.com/trpc/trpc/pull/5165#discussion_r1482661949

## Steps to reproduce

```bash
pnpm i
pnpm vitest
```

You'll get a `fetch failed` error that originates from an invalid `Uint8Array` comparison in `@web3-storage/multipart-parser`, that an `a instanceof Uint8Array` returns `false` when it should be `true`.

```bash
[@web3-storage/multipart-parser]: a =  Uint8Array(34) [
   45,  45,  45, 45,  45, 45, 102, 111,
  114, 109, 100, 97, 116, 97,  45, 117,
  110, 100, 105, 99, 105, 45,  48,  51,
   51,  51,  55, 49,  56, 53,  49,  56,
   49,  52
] a instanceof Uint8Array =  true
[@web3-storage/multipart-parser]: a =  Uint8Array(167) [
   45,  45,  45,  45,  45,  45, 102, 111, 114, 109, 100,  97,
  116,  97,  45, 117, 110, 100, 105,  99, 105,  45,  48,  51,
   51,  51,  55,  49,  56,  53,  49,  56,  49,  52,  13,  10,
   67, 111, 110, 116, 101, 110, 116,  45,  68, 105, 115, 112,
  111, 115, 105, 116, 105, 111, 110,  58,  32, 102, 111, 114,
  109,  45, 100,  97, 116,  97,  59,  32, 110,  97, 109, 101,
   61,  34, 102, 105, 108, 101,  34,  59,  32, 102, 105, 108,
  101, 110,  97, 109, 101,  61,  34,  98, 111,  98,  46, 116,
  120, 116,  34,  13,
  ... 67 more items
] a instanceof Uint8Array =  false
```

This seems to be caused when using the `jsdom` environment in Vitest. Disabling that environment in `vite.config.js` makes the test pass.

```diff
-  environment: "jsdom",
+  // environment: "jsdom",
```

## Expected behavior

All `a instanceof Uint8Array` should return `true` when using the `jsdom` environment, which would make the test pass.

## Additional notes

Maybe this issue should be filed somewhere else, idk. Starting here since the runner is Vitest...

### Reproduction

Maybe not minimal but in somewhat real-world context: https://github.com/juliusmarminge/vitest-repro

### System Info

```shell
System:
    OS: macOS 14.2.1
    CPU: (8) arm64 Apple M1 Pro
    Memory: 70.72 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.11.0 - ~/Library/Caches/fnm_multishells/75747_1707647462712/bin/node
    Yarn: 1.22.19 - ~/Library/Caches/fnm_multishells/75747_1707647462712/bin/yarn
    npm: 10.2.4 - ~/Library/Caches/fnm_multishells/75747_1707647462712/bin/npm
    pnpm: 8.13.1 - ~/Library/Caches/fnm_multishells/75747_1707647462712/bin/pnpm
    bun: 1.0.26 - ~/.bun/bin/bun
  Browsers:
    Chrome: 121.0.6167.160
    Safari: 17.2.1
  npmPackages:
    vite: ^5.1.1 => 5.1.1
    vitest: ^1.2.2 => 1.2.2
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by hi-ogawa** [comment]
From what I investigated so far, the description of "flaky" might not be accurate. One `Uint8Array` is coming from `jsdom` polyfill (which is available globally during test) and another `Uint8Array` is coming from `Readable.toWeb` of `node:stream`:

https://github.com/trpc/trpc/blob/8db7f7635a261cd21e1c046ba22e858feae7ffba/packages/server/src/adapters/node-http/content-type/form-data/index.ts#L48-L49

The first one succeeding `instanceof Uint8Array` check is because the first one happens to be constructed outside of `Readable.toWeb`:

https://github.com/web3-storage/multipart-parser/blob/844cdf7bcb4d86274ae36fce70b1480a78a16774/src/index.js#L146

For some reason, I cannot reproduce on stackblitz but I made a simpler example for demo:
- https://stackblitz.com/edit/github-6mahon?file=repro.test.ts

---

Considering you came to this issue by simply upgrading Vitest, it's still likely that some changes in Vitest affected this, but I couldn't figure out that part yet.

I'm not sure if it's crucial to use jsdom for your test case on trpc, but for the time being, have you considered using `// @vitest-environment node` just for this file?
